src/org/mozilla/javascript/JavaMembers.java

```diff
@@ -918,3 +740,988 @@ public Object getDefaultValue(Class<?> hint)
+                    ((NativeJavaMethod)member).methods.length > 1 ) {
+                    NativeJavaMethod fun =
+                        new NativeJavaMethod(methodOrCtor, name);
+                    fun.setPrototype(prototype);
+                    ht.put(name, fun);
+                    member = fun;
+                }
+            }
+        }
+
+        return member;
+    }
+
+    protected void initField(Scriptable scope, Field field) {
+        String name = field.getName();
+        int mods = field.getModifiers();
+        try {
+            boolean isStatic = Modifier.isStatic(mods);
+            Map<String,Object> ht = isStatic ? staticMembers : members;
+            Object member = ht.get(name);
+            if (member == null) {
+                ht.put(name, field);
+            } else if (member instanceof NativeJavaMethod) {
+                final FieldAndMethods fam = initFieldAndMethods(scope, field, name, isStatic, member);
+                ht.put(name, fam);
+            } else if (member instanceof Field) {
+                Field oldField = (Field) member;
+                // If this newly reflected field shadows an inherited field,
+                // then replace it. Otherwise, since access to the field
+                // would be ambiguous from Java, no field should be
+                // reflected.
+                // For now, the first field found wins, unless another field
+                // explicitly shadows it.
+                if (oldField.getDeclaringClass().
+                        isAssignableFrom(field.getDeclaringClass()))
+                {
+                    ht.put(name, field);
+                }
+            } else {
+                // "unknown member type"
+                Kit.codeBug();
+            }
+        } catch (SecurityException e) {
+            // skip this field
+            Context.reportWarning("Could not access field "
+                    + name + " of class " + cl.getName() +
+                    " due to lack of privileges.");
+        }
+    }
+
+    private FieldAndMethods initFieldAndMethods(Scriptable scope, Field field, String name, boolean isStatic,
+            Object member) {
+        NativeJavaMethod method = (NativeJavaMethod) member;
+        FieldAndMethods fam
+            = new FieldAndMethods(scope, method.methods, field);
+        Map<String,FieldAndMethods> fmht = isStatic ? staticFieldAndMethods
+                                  : fieldAndMethods;
+        if (fmht == null) {
+            fmht = new HashMap<String,FieldAndMethods>();
+            if (isStatic) {
+                staticFieldAndMethods = fmht;
+            } else {
+                fieldAndMethods = fmht;
+            }
+        }
+        fmht.put(name, fam);
+        return fam;
+    }
+
+     private ClassReflectBean reflect(Scriptable scope,
+             boolean includeProtected,
+             boolean includePrivate)
+{
+// We reflect methods first, because we want overloaded field/method
```

Comment:
Looks like we stopped indenting here. The main rule we have in this codebase right now is to keep the code style consistent with whatever else is already in this file, so could you please fix the indenting here and elsewhere?

(I wish that we had strict indenting standards for this project but we're about 20 years too late.)

Reply:
> Looks like we stopped indenting here. The main rule we have in this codebase right now is to keep the code style consistent with whatever else is already in this file, so could you please fix the indenting here and elsewhere?
> 
> (I wish that we had strict indenting standards for this project but we're about 20 years too late.)

I agree , I'll fix later:)

we can do better, why not.